### PR TITLE
Update doc generation with templates and checked via github actions

### PR DESCRIPTION
**Pull request by lambcode**
This PR does the following
1) Updates the docs files to be generated by the latest version of `tfplugindocs`
2) Uses template files to separate the autogenerated portions of the docs files from the description/examples. This makes it much easier to update in the future
3) Adds section the DEVELOPMET.md file on how to update docs files
4) Adds a github action to ensure the checked in docs files matches what would be generated by running `tfplugindocs`

I wouldn't spend too much time reviewing the template or docs files. The changes in the docs directory are mostly due to the newer version of tfplugindocs, and the new files in templates just pull out the non-generated portion of the original docs files.

@mattcarmody @smithclay @ltyson

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -27,6 +27,30 @@ jobs:
     - name: Format all .tf files recursively
       run: |
         terraform fmt -check -diff -recursive ${{ github.workspace }}
+  terraform_docs:
```
Comment (lambcode) [inline .github/workflows/pr.yml:30]: This file is interesting to review.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -27,3 +27,13 @@ provider_installation {
 ```
 
 3) Import the module in another project pinned to the version that matches `.version`. Delete any terraform lock files and run `terraform init`. 
+
+## Updating docs
```
Comment (lambcode) [inline DEVELOPMENT.md:31]: This is the other interesting file to review

File: docs/resources/stream_dashboard.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+---
+Stream Dashboards are deprecated and will be removed shortly. DO NOT USE.
```
Comment (lambcode) [inline docs/resources/stream_dashboard.md:2]: I assume this is true and why the `stream_dashboard.md` file was missing before this PR.

File: templates/resources/alerting_rule.md.tmpl [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "lightstep_alerting_rule Resource - terraform-provider-lightstep"
+subcategory: ""
+description: |-
+
+---
+
+# lightstep_alerting_rule (Resource)
+
+Provides a (Lightstep Alerting Rule)[https://api-docs.lightstep.com/reference/listalertingrules]. This can be used to create and manage Lightstep alerting rules.
```
Comment (mattcarmody) [inline templates/resources/alerting_rule.md.tmpl:11]: is the link syntax backward? ()[] rather than []()

Comment (lambcode) [reply templates/resources/alerting_rule.md.tmpl:11]: Good catch. This is borked in the current provider too. Whoops! https://registry.terraform.io/providers/lightstep/lightstep/latest/docs/resources/alerting_rule

I'll fix

File: .github/workflows/pr.yml [hunk 5]
```diff
@@ -27,6 +27,30 @@ jobs:
...
+          cd /tmp
+          curl -L -o tfplugindocs.zip https://github.com/hashicorp/terraform-plugin-docs/releases/download/v0.9.0/tfplugindocs_0.9.0_linux_amd64.zip
+          unzip tfplugindocs.zip
+          chmod +x tfplugindocs
+
+      - name: Compare generated files with checked in files
+        run: |
+          rm -r docs
+          /tmp/tfplugindocs
+          git diff --stat --exit-code
```
Comment (smithclay) [inline .github/workflows/pr.yml:53]: optional: scope the diff to the `docs` directory: `git diff --stat --exit-code ./docs`

**Review by smithclay — APPROVED** [review]
This looks great. Only comment is an optional change that scopes the diff to the `docs/` directory.

File: docs/resources/stream_dashboard.md (hunk 3 above)
Comment (smithclay) [reply docs/resources/stream_dashboard.md:2]: yes, that's the case.

**Pull request merged**